models/_root/product.yml

```diff
@@ -124,6 +124,13 @@ properties:
     type: integer
   options:
     $ref: ../product/options.yml
+  out_of_stock:
+    type: boolean
+    description: Boolean that indicates whether the product is out of stock.

```

Comment:
Maybe add some clarification here that this only applies to the default selection on page load?
Example, if Product A is out of the "Small" size, but "Small" is selected by default, **out_of_stock** will return true, even if the product is not out of stock as a whole.